SNIPS/snip-9.md

```diff
@@ -0,0 +1,108 @@
+- Fee subsidy: since the sender of the transaction pays gas fees, there’s no need for the account to be funded with any gas tokens. Although this is not the main goal, this proposal is a de-facto solution that already works while paymasters and nonce abstraction are being designed for Starknet.
+
+## Specification
+
+An app can execute “outside transactions” by building a struct representing the execution, having it signed in the user’s wallet, then passing it to a custom method on the account contract.
+
+### 1. Build the `OutsideExecution` struct
+
+An `OutsideExecution` represents a transaction to be executed on behalf of the user account, passed in by another contract. Below is a Cairo representation, but it needs to be constructed offchain.
+
+```rust
+#[derive(Copy, Drop, Serde)]
+struct OutsideExecution {
+    caller: ContractAddress,
+    nonce: felt252,
+    execute_after: u64,
+    execute_before: u64,
+    calls: Span<Call>
+}
+```
+
+- **caller**: can be used to restrict which calling contracts can initiate this execution, although a special address `'ANY_CALLER'` can be used to allow every caller.
+- **nonce**: this is different from the account’s usual nonce, it is used to prevent signature reuse across executions and doesn’t need to be incremental as long as it’s unique.
+- **execute_{before,after}**: timestamp range in which the execution is allowed.
+- **calls**: the usual calls to be executed by the account.
+
+### 2. Sign it using ERC-712 typed data hashing
+
+Reference implementation: [https://github.com/argentlabs/argent-contracts-starknet/blob/main/tests/lib/outsideExecution.ts](https://github.com/argentlabs/argent-contracts-starknet/blob/main/tests/lib/outsideExecution.ts)
+
+See this SNIP for more info on offchain signatures on Starknet: [https://community.starknet.io/t/snip-off-chain-signatures-a-la-eip712/98029](https://community.starknet.io/t/snip-off-chain-signatures-a-la-eip712/98029)
+
+### 3. Pass the struct and signature to the account
+
+Check if the account supports this SNIP:
+
+```rust
+let acccount = IErc165Dispatcher { contract_address: acount_address };
+let is_supported = account.supports_interface(ERC165_OUTSIDE_EXECUTION_INTERFACE_ID); // see below for actual value
+```
+
+Call the `execute_from_outside` method on the account:
+
+```rust
+let acccount = IOutsideExecutionDispatcher { contract_address: acount_address };
+// pre-execution logic...
+let results = account.execute_from_outside(outside_execution, signature);
+// post-execution logic...
+```
+
+### For account buildoors
+
+To accept such outside transactions, the account contract must implement the following interface:
+
+```rust
+/// Interface ID: 0x68cfd18b92d1907b8ba3cc324900277f5a3622099431ea85dd8089255e4181
+#[starknet::interface]
+trait IOutsideExecution<TContractState> {
+    /// @notice This method allows anyone to submit a transaction on behalf of the account as long as they have the relevant signatures
```

Comment:
Move to Rust style doc format please, as we converged to this format

Reply:
moved